### PR TITLE
Fix issue #342

### LINKED ISSUE #342: `traits.testing.unittest_tools._TraitsChangeCollector` overrides the `HasTraits.trait` function
`traits.testing.unittest_tools._TraitsChangeCollector` is a `HasTraits` class that defines an attribute named `trait` which results in replacing the `trait` function of the HatTraits class. The following example demonstrates the issue:

*sample*
```
from traits.api import Float, HasTraits
from traits.testing.unittest_tools import _TraitsChangeCollector

class TestObject(HasTraits):

    number = Float(2.0)

test_object = TestObject()
collector = _TraitsChangeCollector(obj=test_object, trait='number')
collector.event = 6   # This will try to add a new trait named event which will fail
```

*traceback*
```
Exception occurred in traits notification handler.
Please check the log file for details.
Exception occurred in traits notification handler for object: <traits.testing.unittest_tools._TraitsChangeCollector object at 0x0000000004C1E150>, trait: trait_added, old value: <undefined>, new value: event
Traceback (most recent call last):
  File "c:\...\lib\site-packages\traits\trait_notifiers.py", line 340, in __call__
    self.handler( *args )
  File "c:\...\lib\site-packages\traits\has_traits.py", line 1071, in _trait_added_changed
    trait = self.trait( name )
TypeError: 'str' object is not callable
```

### PR DESCRIPTION
**Pull request by itziakos**
This PR renames the `trait` attribute of `_TraitsChangeCollector` to fix #342. There is no change at the constructor signature but I add a deprecation warning that prompts users to stop using `trait`

### CODE REVIEW

**Comment by mdickinson** [comment]
Given that this is a private class, it's tempting not to bother with the `DeprecationWarning`.

OTOH, apparently this class is directly imported in canopy_data and traits_enaml. :-(

**Comment by mdickinson** [comment]
> this class is directly imported in canopy_data and traits_enaml

and pyface, too.

**Comment by mdickinson** [comment]
We should probably make this class public at some point.

LGTM. I'm not a big fan of the name `monitor_traits`. What do you think of using `trait_name` instead?

**Comment by mdickinson** [comment]
Another possibility, which I think I prefer to this one: don't deprecate use of the `trait` keyword at all. Rename the `trait` trait to `_trait`, and do the same dance in `__init__` as before to accept `trait` as a keyword and assign the corresponding object to the `_trait` trait. This lets us use the existing more natural signature when creating a `_TraitsChangeCollector` object, and avoids the need to update other packages that are importing this object.

The downside is that now if you want to dig the `trait` out of the collector, you need to access it through `_trait`. But that doesn't seem like something that would be needed to me.

File: traits/testing/unittest_tools.py [hunk 1]
```diff
@@ -156,16 +156,26 @@ class _TraitsChangeCollector(HasStrictTraits):
     # simultaneously.
     _lock = Instance(threading.Lock, ())
 
+    def __init__(self, **traits):
+        value = traits.pop('trait', None)
+        if value is not None:
```
Comment (corranwebster) [inline traits/testing/unittest_tools.py]: I think `if 'trait' in traits:` is better here with a `traits['monitor_trait'] = traits.pop('trait')` below.

Less confusing if initialized with `trait=None` (which will slip through undetected as an error).

**Comment by corranwebster** [comment]
I agree about the name `monitor_trait`.  I think `trait_name` is preferable to the option of using `_trait`.

**Comment by stefanoborini** [comment]
@corranwebster can this be merged?

File: traits/testing/unittest_tools.py [hunk 2]
```diff
@@ -18,16 +18,16 @@
...
+    Any, Event, HasStrictTraits, Instance, Int, List, Str, Property)
 from traits.util.async_trait_wait import wait_for_condition
 
 # Compatibility layer for Python 2.6: try loading unittest2
 from traits import _py2to3
 if sys.version_info[:2] == (2, 6):
     import unittest2 as unittest
 else:
-    import unittest
+    import unittest  # noqa
```
Comment (itziakos) [inline traits/testing/unittest_tools.py]: Done

File: traits/testing/unittest_tools.py (hunk 1 above)
Comment (itziakos) [inline traits/testing/unittest_tools.py]: done

**Pull request merged**